internal/cmd/commands/connect/funcs.go

```diff
@@ -99,15 +100,60 @@ func fmtSecretForTable(indent int, sc *targets.SessionCredential) []string {
 	if err != nil {
 		return origSecret
 	}
-	dst := new(bytes.Buffer)
-	if err := json.Indent(dst, in, fmt.Sprintf("%s    ", prefixStr), fmt.Sprintf("%s  ", prefixStr)); err != nil {
+
+	// The following operations are performed to better format unspecified credential types.
+	// Credential values are printed with no indentation so that X.509 certificates can be
+	// easily copied and pasted by the user.
+	var out []string
+	var baseCredMap map[string]any
+	err = json.Unmarshal(in, &baseCredMap)
+	if err != nil {
 		return origSecret
 	}
-	secretStr := strings.Split(dst.String(), "\n")
-	if len(secretStr) > 0 {
-		secretStr[0] = fmt.Sprintf("%s    %s", prefixStr, secretStr[0])
+	for k, iface := range baseCredMap {
+		switch iface := iface.(type) {
+		case nil:
+			out = append(out, fmt.Sprintf("%s    %s:", prefixStr, k))
+		case map[string]any:
+			// TODO: check if this is the only possible case for an 'unspecified' type credential
+			// and get rid of the other cases if this is so.
+			out = append(out, fmt.Sprintf("%s    %s:", prefixStr, k))
+			certs := make(map[string]any, len(iface))
```

Comment:
Nit: is there a good reason to bother setting the size of the map here? It seems to me we don't really know how big it's going to be.